ansible/roles/telegraf_systemd/tasks/install_telegraf.yml

```diff
@@ -1,17 +1,20 @@
 ---
 - name: Get influxdata apt repo key
   ansible.builtin.get_url:
-    url: https://repos.influxdata.com/influxdb.key
-    dest: /etc/apt/trusted.gpg.d/influxdb.asc
+    url: https://repos.influxdata.com/influxdata-archive_compat.key
+    dest: /etc/apt/keyrings/influxdata-archive_compat.pgp
     mode: 0644
     owner: root
     group: root
   environment: "{{ proxy_env | default('') }}"
+  register: get_url_output
+  notify: Dearmor pgp key to gpg key
 
 - name: Install influxdata apt repo
   ansible.builtin.apt_repository:
-    repo: "deb https://repos.influxdata.com/{{ ansible_lsb.id | lower }} {{ ansible_lsb.codename }} stable"
-    filename: influxdata
+    repo: "deb [arch=amd64 signed-by={{ influxdata_apt_repo_key }}] \
```

Comment:
This task depends on `influxdata_apt_repo_key`, which is created by a handler. The handler will only be executed at the end, so I think a clean run would fail. You could flush the handler, but in general it doesn't look a very solid approach to me. I'd just have the task execute no matter what. If we want really want idempotency, we can figure out a good expression for `changed_when`, but as a general idea it doesn't look like a handler type of task.

Reply:
I removed the whole handling of dearmoring completely. Apt can also deal with ASCII armored text as opposed to openPGP binary keyring files / gpg files.